aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -161,6 +161,27 @@ Tensor& _clamp_min_out_cpu(Tensor& result, const Tensor& self, Scalar min) {
   return result;
 }
 
+Tensor sign(const Tensor& self) {
+    Tensor result = at::empty({0}, self.options());
+    return at::sign_out(result, self);
+}
+
+Tensor& sign_(Tensor& self) {
+    return at::sign_out(self, self);
+}
+
+Tensor& sign_out(Tensor& result, const Tensor& self) {
+    checkBackend("sign", result, self.type().backend());
+    assert_no_internal_overlap(result);
+    auto iter = TensorIterator::unary_op(result, self);
+    sign_stub(iter.device_type(), iter);
+
+#ifdef BUILD_NAMEDTENSOR
```

Comment:
TensorIterator automatically does names propagation.

Reply:
When I look at UnaryOps.cpp on pytorch/master there are still many of such macro for names propagation, should I remove it totally ?

Reply:
No, please try to keep changes localized to the PR scope.